tests/unittests/instancetype_test.go

```diff
@@ -0,0 +1,220 @@
+package tests
+
+import (
+	"fmt"
+	"strconv"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/resource"
+	instancetypev1beta1 "kubevirt.io/api/instancetype/v1beta1"
+)
+
+const (
+	expectedVersion          = "instancetype.kubevirt.io/v1beta1"
+	expectedVendor           = "kubevirt.io"
+	instanceTypeErrorMessage = "%s label does not match in instancetype: %s"
+	preferenceErrorMessage   = "%s label does not match in preference: %s"
+)
+
+var testsClusterPreferences []instancetypev1beta1.VirtualMachineClusterPreference
+var testsClusterInstanceType []instancetypev1beta1.VirtualMachineClusterInstancetype
+
+var _ = Describe("Common instance types unit tests", func() {
+
+	skipLabels := map[string]bool{
+		"instancetype.kubevirt.io/os-type":                      true,
+		"instancetype.kubevirt.io/deprecated":                   true,
+		"instancetype.kubevirt.io/common-instancetypes-version": true,
+		"instancetype.kubevirt.io/version":                      true,
+		"instancetype.kubevirt.io/class":                        true,
+	}
+
+	instanceTypeFunctionMap := map[string]func(string, string, instancetypev1beta1.VirtualMachineClusterInstancetype) error{
+		"instancetype.kubevirt.io/cpu":                   checkCPU,
+		"instancetype.kubevirt.io/memory":                checkMemory,
+		"instancetype.kubevirt.io/hugepages":             checkHugepages,
+		"instancetype.kubevirt.io/gpus":                  checkGPUs,
+		"instancetype.kubevirt.io/numa":                  checkNuma,
+		"instancetype.kubevirt.io/dedicatedCPUPlacement": checkDedicatedCPUPlacement,
+		"instancetype.kubevirt.io/isolateEmulatorThread": checkIsolateEmulatorThread,
+		"instancetype.kubevirt.io/vendor":                instancetypeCheckVendor,
+	}
+
+	preferenceFunctionMap := map[string]func(string, string, instancetypev1beta1.VirtualMachineClusterPreference) error{
+		"openshift.io/display-name":       checkDisplayName,
+		"instancetype.kubevirt.io/vendor": preferenceCheckVendor,
+	}
+
+	BeforeEach(func() {
+		// Reload to make sure we have unchanged data
+		testsClusterPreferences = virtualMachineClusterPreferences
```

Comment:
This will create a shallow copy of the slice, we need an explicit full copy of the slices.